performance/theme_runner.rb

```diff
@@ -45,22 +45,13 @@ def compile
   end
 
   def run
-    # Dup assigns because will make some changes to them
-    assigns = Database.tables.dup
-
-    @tests.each do |liquid, layout, template_name|
-      # Compute page_tempalte outside of profiler run, uninteresting to profiler
-      page_template = File.basename(template_name, File.extname(template_name))
+    each_test do |liquid, layout, assigns, page_template, template_name|
       compile_and_render(liquid, layout, assigns, page_template, template_name)
     end
   end
 
   def compile_and_render(template, layout, assigns, page_template, template_file)
-    tmpl = Liquid::Template.new
-    tmpl.assigns['page_title'] = 'Page title'
-    tmpl.assigns['template'] = page_template
-    tmpl.registers[:file_system] = ThemeRunner::FileSystem.new(File.dirname(template_file))
-
+    tmpl = init_template(page_template, template_file)
```

Comment:
This could be
```ruby
  def compile_and_render(template, layout, assigns, page_template, template_file)
    compiled_test = compile(template, layout, assigns, page_template, template_file)

    if layout
      assigns['content_for_layout'] = compiled_test[:tmpl].render!(assigns)
      compiled_test[:layout].render!(assigns)
    else
      compiled_test[:tmpl]
    end
  end
```
removing the duplication flo mentioned